aws/resource_aws_wafv2_web_acl_test.go

```diff
@@ -0,0 +1,1128 @@
+					resource.TestCheckResourceAttr(resourceName, "rule.2578204250.override_action.0.count.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2578204250.override_action.0.none.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2578204250.statement.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2578204250.statement.0.managed_rule_group_statement.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2578204250.statement.0.managed_rule_group_statement.0.name", "AWSManagedRulesCommonRuleSet"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2578204250.statement.0.managed_rule_group_statement.0.vendor_name", "AWS"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2578204250.statement.0.managed_rule_group_statement.0.excluded_rule.#", "0"),
+				),
+			},
+			{
+				Config: testAccAwsWafv2WebACLConfig_ManagedRuleGroupStatement_Update(webACLName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsWafv2WebACLExists(resourceName, &v),
+					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "wafv2", regexp.MustCompile(`regional/webacl/.+$`)),
+					resource.TestCheckResourceAttr(resourceName, "name", webACLName),
+					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2461626547.name", "rule-1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2461626547.action.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2461626547.override_action.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2461626547.override_action.0.count.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2461626547.override_action.0.none.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2461626547.statement.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2461626547.statement.0.managed_rule_group_statement.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2461626547.statement.0.managed_rule_group_statement.0.name", "AWSManagedRulesCommonRuleSet"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2461626547.statement.0.managed_rule_group_statement.0.vendor_name", "AWS"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2461626547.statement.0.managed_rule_group_statement.0.excluded_rule.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2461626547.statement.0.managed_rule_group_statement.0.excluded_rule.0.name", "SizeRestrictions_QUERYSTRING"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2461626547.statement.0.managed_rule_group_statement.0.excluded_rule.1.name", "NoUserAgent_HEADER"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testAccAwsWafv2WebACLImportStateIdFunc(resourceName),
+			},
+		},
+	})
+}
+
+func TestAccAwsWafv2WebACL_Minimal(t *testing.T) {
```

Comment:
could we similarly add a test w/empty blocks where possible, i.e. `rule {}`?

Reply:
Empty block is not possible as there are required fields, what would you like to test?

Reply:
i overlooked the first level under `rule` was not fully optional..disregard please!